db/cache/cache.go

```diff
@@ -0,0 +1,139 @@
+package cache
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/livepeer/leaderboard-serverless/common"
+	"github.com/livepeer/leaderboard-serverless/models"
+)
+
+type Cache interface {
+	InvalidateRegionsCache()
+	GetRegions() CacheResult
+	UpdateRegions(newRegions []*models.Region)
+	InvalidatePipelinesCache()
+	GetPipelines() CacheResult
+	UpdatePipelines(newPipelines []*models.Pipeline)
+}
+
+type CacheResult struct {
+	Results      interface{}
+	LastUpdate   time.Time
+	CacheHit     bool
+	CacheExpired bool
+}
+
+type MemCache struct {
+	mu                    sync.RWMutex
+	regionsCacheTimeout   time.Duration
+	regions               []*models.Region
+	regionsLastUpdate     time.Time
+	pipelinesCacheTimeout time.Duration
+	pipelines             []*models.Pipeline
+	pipelinesLastUpdate   time.Time
+}
+
+func NewCache() *MemCache {
```

Comment:
Have you considered using a library instead of implementing the cache?

For example, this one: 
https://github.com/patrickmn/go-cache

Reply:
I did.  Again, we tried to avoid adding new packages to this project when possible.  The goal was to introduce as minimal changes to the footprint as possible and avoid elements that may break the Vercel deployment.